### DISCUSSION TITLE
Different distributions per control variable

### DISCUSSION THREAD

**Discussion by anor4k** [google/meridian#265] [Q&A] [answered]
Hi! I'm trying to understand how I could pass different Prior distributions to each of my control variables. Something like:

```python
prior = md.model.prior_distribution.PriorDistribution(
    gamma_c=[
      tfp.distributions.half_normal.Normal(0, 1),
      tfp.distributions.half_normal.HalfNormal(5),
    ],
)
```

But the above sample doesn't work, since it expects a single distribution.

I tried using the ` tfp.distributions.independent_joint_distribution_from_structure` function from tensorflow probability but no luck.

**Comment by alexwakim** [comment, marked as answer]
Hi, we don't currently support mixing distributions for a single parameter.

Can you use [tfp.distriubtions.TruncatedNormal](https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/TruncatedNormal#:~:text=The%20truncated%20normal%20is%20a,this%20implementation%20is%20fully%20reparameterized.) to get something close to what you want? You can use different points of truncation.

**Reply by anor4k** [reply]
Yes, using `TruncatedNormal` seemed to be the best alternative! Thanks.
